### ISSUE TITLE
predict_proba method in the pipeline

### ISSUE THREAD

**Issue by ravinkohli** [LMZimmer/Auto-PyTorch_refactor#55]
**Current Problem**
The current predict_proba in autopytorch base pipeline calls the predict_proba of the sklearn base pipeline, which calls the predict_proba of the last step in the pipeline, i.e, the TrainerChoice, which doesn't have a predict_proba method. Therefore, it gives an error. 

**Suggestion**
as for the base_pipeline where we have defined a predict method which calls the predict on the network, we should also have a predict_proba in which we run our decision through a softmax and return it.

**Comment by ravinkohli** [comment]
Solved in #54
